Rakefile

```diff
@@ -15,14 +15,133 @@
 ##########################################################################
 
 require "fileutils"
+require 'json'
+require 'open-uri'
+require_relative 'scripts/load_scenarios'
+require_relative 'scripts/init'
 
-task :start do
-  ENV['URL'] = "http://localhost:8153"
-  ENV['JMETER_PATH'] = "/usr/local/Cellar/jmeter/2.13/bin/"
+task :start_stop_perf do
+  begin
+    cleanup
+    @scm_pid = scm_commit_loop
+    @config_pid = config_update_loop
+    warm_up
+    ruby "scripts/run_test.rb"
+  ensure
+    p "Starting clean up Process for #{@scm_pid} and #{@config_pid}"
+    destroy @scm_pid if defined? @scm_pid
+    destroy @config_pid if defined? @config_pid
+    stop_agents
+    git_cleanup
+  end
+end
+
+def cleanup
   File.delete('jmeter.jmx') if File.exists?('jmeter.jmx')
   File.delete('jmeter.log') if File.exists?('jmeter.log')
   File.delete('custom.log') if File.exists?('custom.log')
-  File.delete('jmeter.jtl') if File.exists?('jmeter.jtl')
   File.delete('perf.jtl') if File.exists?('perf.jtl')
-  ruby "scripts/run_test.rb"
+  File.delete('jmeter.jtl') if File.exists?('jmeter.jtl')
+end
+
+def scm_commit_loop
```

Comment:
Pull this out into a common function and pass the command to be executed in the loop with an interval
https://github.com/gocd/support-and-security/blob/37003b221c5c37ef12eb31c4573d04e1fa5c9554/support/analyze#L14